src/main/java/de/tum/cit/aet/artemis/exercise/domain/Exercise.java

```diff
@@ -847,17 +847,18 @@ && isValidExampleSolutionPublicationDate(getStartDate(), getDueDate(), getExampl
 
     /**
      * Validates score settings
-     * 1. The maxScore needs to be greater than 0
+     * 1. The maxScore needs to be greater than 0. If maxScore is null or <= 0, it is set to the default value 1.0.
      * 2. If the specified amount of bonus points is valid depending on the IncludedInOverallScore value
      */
     public void validateScoreSettings() {
         // Check if max score is set
         if (getMaxPoints() == null || getMaxPoints() <= 0) {
-            throw new BadRequestAlertException("The max points needs to be greater than 0", "Exercise", "maxScoreInvalid");
+            // Correct invalid maxPoints to default value (prevents invalid state)
+            setMaxPoints(1.0);
```

Comment:
I don't think this is a good change. It's intended that the method throws an exception when validating incoming data from the client. If you change this like that, the server would accept it, but change the value provided by the client which is not a good idea